libsolidity/ast/Types.cpp

```diff
@@ -2400,7 +2400,10 @@ MemberList::MemberMap StructType::nativeMembers(ASTNode const*) const
 	{
 		TypePointer type = variable->annotation().type;
 		solAssert(type, "");
-		solAssert(!(location() != DataLocation::Storage && type->containsNestedMapping()), "");
+		// If we are not in storage, skip all members that cannot live outside of storage,
```

Comment:
Previously  we allowed structs containing mappings in memory and skipped such members when this function was being called. Then the TypeChecker dissalowed such structs. Do you think you can keep the old code and add a check about constants?